contracts/src/instances/lseth/LsETHBase.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IRiverV1 } from "../../interfaces/lseth/IRiverV1.sol";
+import { HyperdriveBase } from "../../internal/HyperdriveBase.sol";
+import { FixedPointMath, ONE } from "../../libraries/FixedPointMath.sol";
+
+/// @author DELV
+/// @title LsETHHyperdrive
+/// @notice The base contract for the lsETH Hyperdrive implementation.
+/// @dev River has it's own notion of shares to account for the accrual of
+///      interest on the ether pooled in the River protocol. Instead of
+///      maintaining a balance of shares, this integration can simply use River's
+///      shares directly.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract LsETHBase is HyperdriveBase {
+    using FixedPointMath for uint256;
+
+    /// @dev The lsETH contract.
+    IRiverV1 internal immutable _river;
+
+    /// @notice Instantiates the lsETH Hyperdrive base contract.
+    /// @param __river The lsETH contract.
+    constructor(IRiverV1 __river) {
+        _river = __river;
+    }
+
+    /// Yield Source ///
+
+    /// @dev Accepts a deposit from the user in base.
```

Comment:
Let's add a sentence about how this option is unsupported for this yield source.